### DISCUSSION TITLE
Designing for Realtime progress update

### DISCUSSION THREAD

**Discussion by ankitv89** [Sunbird-Lern/Community#172] [Query]
Use Case:

At Aastrika we use SCORM content for most of our learning resources. We use an adapter that reads progress from Scorm and accordingly updates the content state. The objective is that users should start from the last read content whenever they click on the resume from the Table of Content page.  

User Jounery--> User starts to consume SCORM resources and exits to the TOC page. Then click on resume and they land at the last consumed resource. Currently, we use the lastContentReadId of enrollment Api to direct the user to the last consumed resource.

Problem area- The content update is processed through the activity aggregator which takes some time to update the enrollment table. If users click on the resume in between they land at the old content they consumed rather than the last one.

Help Sought: Suggestion to get the progress/last used resource on a real-time basis.

**Comment by AmiableAnil** [comment]
@ankitv89 As of now we are syncing it asynchronously. We can have a call to check the possibility of making it as sync process.
